root/io/transient/base/CMakeLists.txt

```diff
@@ -2,7 +2,11 @@
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 set(CMAKE_ROOTTEST_NOROOTMAP OFF)
 ROOT_GENERATE_DICTIONARY(base_cling ${CMAKE_CURRENT_SOURCE_DIR}/base.h MODULE base LINKDEF baseLinkDef.h)
-ROOTTEST_LINKER_LIBRARY(base base_cling.cxx testobject.cpp testobjectderived.cpp TEST LIBRARIES Core RIO)
+if(MSVC AND DEFINED ROOT_SOURCE_DIR)
+  ROOTTEST_LINKER_LIBRARY(base base_cling.cxx testobject.cpp testobjectderived.cpp TEST LIBRARIES Core RIO)
+else()
+  ROOTTEST_LINKER_LIBRARY(base base_cling.cxx testobject.cpp testobjectderived.cpp TEST LIBRARIES ROOT::Core ROOT::RIO)
+endif()
```

Comment:
It might make sense to do add a `ROOTTEST_LINK_LIBRARIES` function, do this logic (i.e. ROOT::LIB vs LIB) inside it, and then just replace `target_link_libraries` with `ROOTTEST_LINK_LIBRARIES` where needed. Then you have a single place where you can control this behavior rather than have many places to remember to change/fix when needed.

Reply:
That said, what you have now should be fine to merge, if you just want to save on build time.

Reply:
For the other places too I suppose (like `ROOT_ADD_GTEST` and `ROOTTEST_GENERATE_EXECUTABLE`)? OK, I'll check and review the whole code.

Reply:
> That said, what you have now should be fine to merge, if you just want to save on build time.

Thanks @amadio ! I'll merge this PR and I'll review the code as we discussed